app/javascript/serviceworker/cache.spec.ts

```diff
@@ -1,49 +1,57 @@
 require("jest-fetch-mock").enableMocks();
 import { addAll, match, put } from "./cache";
-import { add } from "./store";
-
-jest.mock("./store");
-
-const addAllMock = jest.fn();
-const matchMock = jest.fn();
-const putMock = jest.fn();
-global.caches = {
-  open: jest.fn(() => ({
-    addAll: addAllMock,
-    match: matchMock,
-    put: putMock,
-  })),
-} as any;
+import { add, getByUrl } from "./store";
 
 jest.mock("./store");
 
 describe("addAll", () => {
   test("works", async () => {
-    await addAll(["foo"]);
-    expect(addAllMock).toHaveBeenCalledWith(["foo"]);
+    const urls = ["https://example.com/test", "https://example.com/test2"];
+    const firstResponse = new Response("foo");
+    const firstBlob = await firstResponse.clone().blob();
+    const secondResponse = new Response("bar");
+    const secondBlob = await secondResponse.clone().blob();
+
+    (fetch as jest.Mock)
+      .mockResolvedValueOnce(firstResponse)
+      .mockResolvedValueOnce(secondResponse);
+    await addAll(urls);
+
+    expect(add).toHaveBeenCalledWith("cachedResponses", urls[0], firstBlob);
+    expect(add).toHaveBeenCalledWith("cachedResponses", urls[1], secondBlob);
   });
 });
 
 describe("match", () => {
   test("works", async () => {
```

Comment:
If this were rspec, I would be pickier about the use of "works" here. Is now a good time to improve this?